debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresChangeEventSourceCoordinator.java

```diff
@@ -0,0 +1,76 @@
+import io.debezium.pipeline.ErrorHandler;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.metrics.spi.ChangeEventSourceMetricsFactory;
+import io.debezium.pipeline.source.spi.ChangeEventSource.ChangeEventSourceContext;
+import io.debezium.pipeline.source.spi.ChangeEventSourceFactory;
+import io.debezium.pipeline.source.spi.SnapshotChangeEventSource;
+import io.debezium.pipeline.spi.OffsetContext;
+import io.debezium.schema.DatabaseSchema;
+
+public class PostgresChangeEventSourceCoordinator extends ChangeEventSourceCoordinator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresChangeEventSourceCoordinator.class);
+
+    private final Snapshotter snapshotter;
+    private final SlotState slotInfo;
+
+    public PostgresChangeEventSourceCoordinator(OffsetContext previousOffset, ErrorHandler errorHandler,
+                                                Class<? extends SourceConnector> connectorType,
+                                                CommonConnectorConfig connectorConfig,
+                                                ChangeEventSourceFactory changeEventSourceFactory,
+                                                ChangeEventSourceMetricsFactory changeEventSourceMetricsFactory,
+                                                EventDispatcher<?> eventDispatcher, DatabaseSchema<?> schema,
+                                                Snapshotter snapshotter, SlotState slotInfo) {
+        super(previousOffset, errorHandler, connectorType, connectorConfig, changeEventSourceFactory, changeEventSourceMetricsFactory, eventDispatcher, schema);
+        this.snapshotter = snapshotter;
+        this.slotInfo = slotInfo;
+    }
+
+    @Override
+    protected CatchUpStreamingResult executeCatchUpStreaming(OffsetContext previousOffset, ChangeEventSourceContext context,
+                                                             SnapshotChangeEventSource snapshotSource)
+            throws InterruptedException {
+        if (previousOffset != null && !snapshotter.shouldStreamEventsStartingFromSnapshot() && slotInfo != null) {
+            try {
+                setSnapshotStartLSN((PostgresSnapshotChangeEventSource) snapshotSource,
+                        (PostgresOffsetContext) previousOffset);
+            }
+            catch (SQLException e) {
+                throw new InterruptedException("Failed to determine catch-up streaming stopping LSN");
+            }
+            LOGGER.info("Previous connector state exists and will stream events until {} then perform snapshot",
+                    LogSequenceNumber.valueOf(((PostgresOffsetContext) previousOffset).getStreamingStoppingLsn()));
+            streamEvents(previousOffset, context);
+            return new CatchUpStreamingResult(true);
+        }
+
+        return new CatchUpStreamingResult(false);
+    }
+
+    private void setSnapshotStartLSN(PostgresSnapshotChangeEventSource snapshotSource,
```

Comment:
```suggestion
    private void setSnapshotStartLsn(PostgresSnapshotChangeEventSource snapshotSource,
```

(we generally prefer this style of abbreviations in identifiers)